qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -112,13 +112,18 @@ def _substitution_checks(self, dag, old_run, new_circ, basis, qubit):
         # if we're outside of the basis set, we're obligated to logically decompose.
         # if we're outside of the set of gates for which we have physical definitions,
         #    then we _try_ to decompose, using the results if we see improvement.
+        new_error = _error(new_circ, self._target, qubit)
```

Comment:
Not sure if it's overkill, but I think it might be cleaner and less error-prone to add a local class to encapsulate the error tuple, and then generate a total ordering for it with `@functools.total_ordering`.

It looks like `_error` is used by `_resynthesize_run` as a key to `min`, so I'd worry there might be times where code accidentally compares tuples to numbers.

Reply:
`Tuple[T, ...]` has an automatic total ordering in Python if type `T` has a total ordering itself, regardless of whether the lengths match - using tuples in comparisons is fairly idiomatic (e.g. `if sys.version_info < (3, 10)` etc).  Maybe I'm misunderstanding what you meant, though.